src/components/NewMessageModal.vue

```diff
@@ -181,6 +182,17 @@ export default {
 				showError(t('mail', 'Could not discard message'))
 			}
 		},
+		async onClose() {
+			// Warning: Ultra hacky! At the moment, there is no other way because the message state
+			// is managed by the Composer component. Otherwise, we could just do something like
+			// this.sendMessage(this.data) or this.saveDraft(this.data).
+			if (this.composerMessage.type === 'outbox') {
+				await this.$refs.composer.onSend()
+			} else {
+				await this.$refs.composer.onSave()
+				this.$emit('close')
```

Comment:
this should be done in both branches, right?

maybe even put a try-catch-finally around to always close and show a toast for a possible error

Reply:
The close is already emitted in the `saveMessage(...)` method. Error handling is a good point. I'll have to test a bit more.

Reply:
I added some toast and more logging in case an error happens. I also wrapped the whole if-else block inside a try/finally block because the modal wouldn't get closed when the sending failed.